rustup-init.ps1

```diff
@@ -0,0 +1,174 @@
+#!/usr/bin/env powershell
+
+# This is just a little script that can be downloaded from the internet to
+# install rustup. It just does platform detection, downloads the installer and
+# runs it.
+
+#XXX: If you change anything here, please make the same changes in setup_mode.rs
+function Write-Usage {
+    Write-Host @'
+rustup-init 1.18.3 (435397f48 2019-05-22)
+The installer for rustup
+
+USAGE:
+    rustup-init.exe [FLAGS] [OPTIONS]
```

Comment:
Is this correct? Shouldn't it be 'rustup-init.ps1 [FLAGS] [OPTIONS]'?

Reply:
Ah right, this had me confused for a moment when I was reading the [`rustup-init.sh`](https://github.com/rust-lang/rustup.rs/blob/master/rustup-init.sh) implementation. It turns out that the usage in both scripts are copied from the output of calling the executable with `rustup-init --help`. I'm pretty sure that this was so that users could get the full set of flags and options without having to download the full binary first. The first iteration of this script was only focused on Windows installations (hence the `.exe` on the end), but it ended up being really easy to support macOS and Linux so that `.exe` gives me squinty eyes now… However, given more Windows users would see this output, I suppose that seeing `.exe` in a principle of least surprise situation?

Reply:
Mmm, really that should be `.ps1` and I ought to ensure that `rustup-init.sh` says `rustup-init.sh` :D

Reply:
Okay, I'll update that--makes sense